src/App.tsx

```diff
@@ -19,34 +24,67 @@ function App() {
     }
   };
 
+  const [showModal, setShowModal] = useState(false);
+
+  const openModal = () => setShowModal(true);
+  const closeModal = () => setShowModal(false);
+
   return (
     <>
-      <AuthCard actionType={authActionType} />
-
-      {/* debug stuff. this is not part of the Auth component */}
-      <Stack gap={3} direction="horizontal" className="mt-5">
-        <Button
-          onClick={() => {
-            authActionType === "signup"
-              ? setAuthActionType("signin")
-              : setAuthActionType("signup");
-          }}
-        >
-          Toggle Form Type
-        </Button>
-        <Button onClick={logoutUser}>Sign Out</Button>
-        <Button
-          onClick={() =>
-            alert(
-              auth.currentUser?.email
-                ? auth.currentUser?.email
-                : "You're not logged in!"
-            )
-          }
-        >
-          Who am I?
-        </Button>
-      </Stack>
+      <Tabs
+        defaultActiveKey="profile"
+        id="uncontrolled-tab-example"
+        className="mb-3"
+      >
+        <Tab eventKey="home" title="Home">
+          <>
+            <Button variant="primary" onClick={openModal}>
+              Sign In
+            </Button>
+
+            <Modal show={showModal} onHide={closeModal}>
+              <Modal.Dialog>
+                <Modal.Header closeButton>
+                  <Modal.Title>Login Modal</Modal.Title>
+                </Modal.Header>
+
+                <Modal.Body>
+                  <AuthCard actionType={authActionType} />
+                </Modal.Body>
+
+                <Modal.Footer>
+                  <Button
+                    onClick={() => {
+                      authActionType === "signup"
+                        ? setAuthActionType("signin")
+                        : setAuthActionType("signup");
+                    }}
+                  >
+                    Toggle Form Type
+                  </Button>
+                  <Button onClick={logoutUser}>Sign Out</Button>
+                  <Button
+                    onClick={() =>
+                      alert(
+                        auth.currentUser?.email
+                          ? auth.currentUser?.email
+                          : "You're not logged in!"
+                      )
+                    }
+                  >
+                    Who am I?
+                  </Button>
+                  <Button variant="secondary" onClick={closeModal}>
+                    Close
+                  </Button>
```

Comment:
```suggestion
```
There's already a close button in the title bar of the modal, so this probably isn't necessary